include/nbl/video/IPhysicalDevice.h

```diff
@@ -328,6 +352,24 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             uint16_t transferDst : 1u;
             uint16_t log2MaxSamples : 3u; // 0 means cant use as a multisample image format
 
+            SFormatImageUsage()
+                : isInitialized(0)
+            {}
+
+            SFormatImageUsage(core::bitflag<asset::IImage::E_USAGE_FLAGS> usages)
+                : isInitialized(1), 
+                sampledImage(usages.hasFlags(asset::IImage::EUF_SAMPLED_BIT)),
+                storageImage(usages.hasFlags(asset::IImage::EUF_STORAGE_BIT)),
+                transferSrc(usages.hasFlags(asset::IImage::EUF_TRANSFER_SRC_BIT)),
+                transferDst(usages.hasFlags(asset::IImage::EUF_TRANSFER_DST_BIT)),
+                attachment((usages& core::bitflag<asset::IImage::E_USAGE_FLAGS>(asset::IImage::EUF_COLOR_ATTACHMENT_BIT | asset::IImage::EUF_DEPTH_STENCIL_ATTACHMENT_BIT)).value != 0),
+                // Other way of getting these flags?
+                blitSrc(usages.hasFlags(asset::IImage::EUF_TRANSFER_SRC_BIT)),
+                blitDst(usages.hasFlags(asset::IImage::EUF_TRANSFER_DST_BIT)),
+                storageImageAtomic(usages.hasFlags(asset::IImage::EUF_STORAGE_BIT)),
+                attachmentBlend((usages& core::bitflag<asset::IImage::E_USAGE_FLAGS>(asset::IImage::EUF_COLOR_ATTACHMENT_BIT | asset::IImage::EUF_DEPTH_STENCIL_ATTACHMENT_BIT)).value != 0)
```

Comment:
only check color attachment bit, only colors can blend...

Reply:
Ok same thing as https://github.com/Devsh-Graphics-Programming/Nabla/pull/375/commits/eda7c31cfebb608c84b3eec345f5ba89499ab82d#r912939223

Leave the things you cannot deduce 100% certain as false, let user patch them up later.